From 9adca474849433ea726ec084a870fb4f2d5b9d7f Mon Sep 17 00:00:00 2001
From: Jerome Brunet <jbrunet@baylibre.com>
Date: Tue, 14 Apr 2020 10:45:08 +0200
Subject: [PATCH 089/118] WIP: clk: meson: g12a: fix hifi pll lock

The HIFI pll of the g12a sometimes takes a long time to report the lock in
HIFI_PLL_CNTL0 bit 31. The would eventually be reported but the delay may
be so long that the driver consider it a lock failure.

Bit 30 seems to do the same job but more quickly, let's try this instead.

Fixes: 085a4ea93d54 ("clk: meson: g12a: add peripheral clock controller")
Signed-off-by: Jerome Brunet <jbrunet@baylibre.com>
---
 drivers/clk/meson/g12a.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/meson/g12a.c b/drivers/clk/meson/g12a.c
index 6d78640ec331..bb4fa19442be 100644
--- a/drivers/clk/meson/g12a.c
+++ b/drivers/clk/meson/g12a.c
@@ -1782,7 +1782,7 @@ static struct clk_regmap g12a_hifi_pll_dco = {
 		},
 		.l = {
 			.reg_off = HHI_HIFI_PLL_CNTL0,
-			.shift   = 31,
+			.shift   = 30,
 			.width   = 1,
 		},
 		.rst = {
-- 
2.31.1

